src/api/vaultDetails.ts

```diff
@@ -29,59 +24,67 @@ async function extractVaultProperties(connector: StakewiseConnector, vault: Hex)
     dateBeforeYesterday.setDate(dateBeforeYesterday.getDate() - 8);
     today.setDate(today.getDate());
 
-    const maybeVault = connector.graphqlRequest({
-        type: 'graph',
-        op: 'Vault',
-        query: `
-        query Vault($address: ID!) {
-          vault(id: $address) {
-            address: id
-            performance: score
-            admin
-            isErc20
-            imageUrl
-            capacity
-            mevEscrow
-            isPrivate
-            createdAt
-            mevEscrow
-            tokenName
-            feePercent
-            totalAssets
-            displayName
-            description
-            whitelister
-            keysManager
-            tokenSymbol
-            feeRecipient
-            validatorsRoot
-            weeklyApy
-          }
-          privateVaultAccounts(
-            where: { vault: $address }
-          ) {
-            createdAt
-            address
-          }
-        }`,
-        variables: vars_getVault,
-        onSuccess: function (value: Response): Response {
-            return value;
-        },
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        onError: function (reason: any): PromiseLike<never> {
-            throw new Error(`Failed to get vault from Stakewise: ${reason}`);
-        },
-    });
+    try {
+        const responseVault = await connector.graphqlRequest({
+            type: 'graph',
+            op: 'Vault',
+            query: `
+            query Vault($address: ID!) {
+              vault(id: $address) {
+                address: id
+                performance: score
+                admin
+                isErc20
+                imageUrl
+                capacity
+                mevEscrow
+                isPrivate
+                createdAt
+                mevEscrow
+                tokenName
+                feePercent
+                totalAssets
+                displayName
+                description
+                whitelister
+                keysManager
+                tokenSymbol
+                feeRecipient
+                validatorsRoot
+                weeklyApy
+              }
+              privateVaultAccounts(
+                where: { vault: $address }
+              ) {
+                createdAt
+                address
+              }
+            }`,
+            variables: vars_getVault,
+            onSuccess: function (value: Response): Response {
+                return value;
+            },
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            onError: function (reason: any): PromiseLike<never> {
+                throw new Error(`Failed to get vault from Stakewise: ${reason}`);
+            },
+        });
+        if (!responseVault.ok) {
```

Comment:
this and next two calls needs to be extracted into some common method, see comment https://github.com/ChorusOne/opus-pool-sdk/pull/16/files#r1486368244